code/modules/admin/verbs/randomverbs.dm

```diff
@@ -1020,6 +1020,18 @@ Traitors and the like can also be revived with the previous role mostly intact.
 	dat += "<br><a href='?src=[REF(src)];[HrefToken()];add_station_goal=1'>Add New Goal</a>"
 	usr << browse(dat, "window=goals;size=400x400")
 
+proc/immerse_player(mob/living/carbon/target, toggle=TRUE, remove=FALSE)
+	var/already_immersed = target.GetComponent(/datum/component/immersion)
+
+	if(already_immersed && (toggle || remove))
+		qdel(already_immersed)
+	else
+		target.AddComponent(/datum/component/immersion)

```

Comment:
Order of this doesn't really make sense. Should be more like this:
Could be *slightly* more sane since it does call ``qdel(null)`` sometimes but still better than currently since yours can ADD the component even IF remove == TRUE.
```
	if(remove || (toggle && already_immersed))
		qdel(already_immersed)
	else
		target.AddComponent(/datum/component/immersion)
```